src/ripple/nodestore/impl/DatabaseShardImp.cpp

```diff
@@ -144,40 +150,46 @@ DatabaseShardImp::init()
                     continue;
                 }
 
-                if (shard->isFinal())
-                {
-                    shards_.emplace(
-                        shardIndex,
-                        ShardInfo(std::move(shard), ShardInfo::State::final));
-                }
-                else if (shard->isBackendComplete())
+                switch (shard->getState())
                 {
-                    auto const result{shards_.emplace(
-                        shardIndex,
-                        ShardInfo(std::move(shard), ShardInfo::State::none))};
-                    finalizeShard(
-                        result.first->second, true, lock, boost::none);
-                }
-                else
-                {
-                    if (acquireIndex_ != 0)
-                    {
+                    case Shard::State::final:
+                        if (++openFinals > openFinalLimit_)
```

Comment:
Just a thought - at this point in the code, would it be better to close final shards by default, and only open later them on demand? At this point we don't know which shards should be prioritized for opening since we don't have the last use time. If we open a bunch of shards that we don't need, those will end up just getting closed later anyway as we keep open the shards we actually need.

Reply:
Sounds good